### PR TITLE
RavenDB-8521 Delayed external replication

### PR DESCRIPTION
**Pull request by karmeli87**

### CODE REVIEW

File: src/Raven.Client/ServerWide/ExternalReplication.cs [hunk 1]
```diff
@@ -13,6 +13,7 @@ public class ExternalReplication : ReplicationNode, IDatabaseTask, IDynamicJsonV
         public string Name;
         public string ConnectionStringName;
         public string MentorNode;
+        public long DelayReplicationFor;
```
Comment (ayende) [inline src/Raven.Client/ServerWide/ExternalReplication.cs:16]: I don't like this data type. Delay replication for what? Ticks, seconds, minutes, years?
Better to use TimeSpan here.

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs [hunk 2]
```diff
@@ -156,10 +158,19 @@ private IEnumerable<ReplicationBatchItem> GetDocsConflictsTombstonesRevisionsAnd
...
         {
             EnsureValidStats(stats);
 
+            long delayReplicationFor = 0;
+            var wasDelayed = false;
+            
+            if (_parent._parent._server.LicenseManager.GetLicenseStatus().HasDelayedExternalReplication 
+                && _parent.Destination is ExternalReplication external)
+            {
+                delayReplicationFor = external.DelayReplicationFor;
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:171]: I think this should error in this case.

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs [hunk 3]
```diff
@@ -221,7 +242,8 @@ public bool ExecuteReplicationOnce(OutgoingReplicationStatsScope stats)
                         // the last etag they have from us on the other side
                         _parent._lastSentDocumentEtag = _lastEtag;
                         _parent._lastDocumentSentTime = DateTime.UtcNow;
-                        _parent.SendHeartbeat(DocumentsStorage.GetDatabaseChangeVector(documentsContext));
+                        var changeVector = wasDelayed ? null : DocumentsStorage.GetDatabaseChangeVector(documentsContext);
+                        _parent.SendHeartbeat(changeVector);
```
Comment (ayende) [inline src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:246]: Shouldn't we send the last change vector we scanned here? (Up to the item we stopped for being too early)

**Review by ayende — CHANGES_REQUESTED** [review]

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs (hunk 3 above)
Comment (karmeli87) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:246]: If we have documents to send, but were delayed (so in this particular case we have _nothing_ to send), we will be sending the current change vector without sending the actual documents.

File: src/Raven.Client/ServerWide/ExternalReplication.cs (hunk 1 above)
Comment (karmeli87) [reply src/Raven.Client/ServerWide/ExternalReplication.cs:16]: Seconds would be ok? so we can also test it in reasonable time.

Comment (ayende) [reply src/Raven.Client/ServerWide/ExternalReplication.cs:16]: Any reasons _not_ to use timespan?

File: src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs (hunk 3 above)
Comment (ayende) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:246]: What if we filtered documents and still have things to send?

Comment (karmeli87) [reply src/Raven.Server/Documents/Replication/ReplicationDocumentSender.cs:246]: So we are not ending in this code path. This is only if we have nothing to send

**Pull request merged**